### PR TITLE
 [C API Docs] Add docs for run options tag/log level accessors/modifiers.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Add documentation for these C API functions:
RunOptionsGetRunLogSeverityLevel
RunOptionsGetRunLogVerbosityLevel
RunOptionsGetRunTag
RunOptionsSetRunLogSeverityLevel
RunOptionsSetRunLogVerbosityLevel
RunOptionsSetRunTag

Update some existing documentation.

**Motivation and Context**
Documentation.

### CODE REVIEW

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 1]
```diff
@@ -474,9 +478,9 @@ struct OrtApi {
   /// @{
 
   /** \brief Create an OrtEnv
-  * 
-  * \param[in] logging_level
-  * \param[in] logid
+  *
+  * \param[in] logging_level The log severity level.
```
Comment (RyanUnderhill) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:482]: The comment makes me think the parameter should be severity_level vs logging_level. Not sure we can really fix it based on the type name.

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 2]
```diff
@@ -474,9 +478,9 @@ struct OrtApi {
...
+  * \param[in] logid The top-level log identifier.
```
Comment (RyanUnderhill) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:483]: What is the top-level log?

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 3]
```diff
@@ -485,10 +489,10 @@ struct OrtApi {
 
   /** \brief Create an OrtEnv
   *
-  * \param[in] logging_function
-  * \param[in] logger_param
-  * \param[in] logging_level
-  * \param[in] logid
+  * \param[in] logging_function A user-provided logging function.
```
Comment (RyanUnderhill) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:492]: All parameters are user provided, maybe 'a pointer to a function called on every logging event'?

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 4]
```diff
@@ -485,10 +489,10 @@ struct OrtApi {
...
+  * \param[in] logger_param Arbitrary user-provided parameter to pass to `logging_function`.
```
Comment (RyanUnderhill) [inline include/onnxruntime/core/session/onnxruntime_c_api.h:493]: Maybe 'A pointer passed as the `param` parameter to the `logging_function` function.'?

File: include/onnxruntime/core/session/onnxruntime_c_api.h (hunk 2 above)
Comment (edgchen1) [reply include/onnxruntime/core/session/onnxruntime_c_api.h:483]: There's some hierarchy - this, then a session log id, and then a run log id. Any suggestion for a better way to put it?

Comment (edgchen1) [reply include/onnxruntime/core/session/onnxruntime_c_api.h]: Well, I guess it's scoped to the environment instance. Won't be more specific here for consistency with other similar parameter descriptions.

**Review by RyanUnderhill — APPROVED** [review]

**Pull request merged**